File: opspilot-enterprise/packages/shared-types/src/approval.ts
export type ApprovalStatus = "pending" | "approved" | "rejected" | "expired" | "recalled";
export type ApprovalRiskLevel = "low" | "medium" | "high" | "critical";
export type ApprovalActionType = "vm_migrate" | "vm_power_off" | "snapshot_delete" | "config_change" | "script_exec" | "upgrade" | "rollback" | "other";

export interface ApprovalRequest {
  id: string;
  title: string;
  description: string;
  action_type: ApprovalActionType;
  risk_level: ApprovalRiskLevel;
  risk_score: number;
  status: ApprovalStatus;
  requester: string;
  assignee: string | null;
  incident_ref: string | null;
  change_analysis_ref: string | null;
  target_object: string;
  target_object_type: string;
  created_at: string;
  updated_at: string;
  expires_at: string | null;
  decision_comment: string | null;
  decided_at: string | null;
  decided_by: string | null;
  tags: string[];
}

export interface ApprovalDecision {
  request_id: string;
  decision: "approved" | "rejected";
  comment: string;
  decided_by: string;
  decided_at: string;
}

File: packages/shared-schema/src/opspilot_schema/approval.py
from __future__ import annotations
from typing import Optional
from pydantic import BaseModel


class ApprovalRequest(BaseModel):
    id: str
    title: str
    description: str
    action_type: str
    risk_level: str
    risk_score: int
    status: str
    requester: str
    assignee: Optional[str] = None
    incident_ref: Optional[str] = None
    change_analysis_ref: Optional[str] = None
    target_object: str
    target_object_type: str
    created_at: str
    updated_at: str
    expires_at: Optional[str] = None
    decision_comment: Optional[str] = None
    decided_at: Optional[str] = None
    decided_by: Optional[str] = None
    tags: list[str] = []


class ApprovalDecision(BaseModel):
    request_id: str
    decision: str
    comment: str
    decided_by: str
    decided_at: str
